### PR TITLE
Get case_id from Pub/Sub message metadata

### PR DESCRIPTION
**Pull request by jcox-dev**
## Motivation and Context
eQ will now pass case_id to metadata and use the transaction id as the filename (and subsequent GCS object id)

## What has changed

- Updated readme with example including custom metadata to file upload
- Updated publish script
- Changed where callback expects to find case_id

## How to test?
Follow [README](https://github.com/ONSdigital/census-rm-pubsub/blob/be95dec61fd041fc8551ca7f6aa6da7bd0de1b49/README.md)

## Links
Waiting on v3 eq Survey Runner changes [in submitter](https://github.com/ONSdigital/eq-survey-runner/blob/113a3aa81418c01fc4a4ed3621d1f827abe3c831/app/submitter/submitter.py#L34)

### CODE REVIEW

**Review by AdamHawtin — APPROVED** [review]
Looks good, tested it locally and the messages on the case responses queue had the correct case ID.

**Comment by jcox-dev** [comment]
Unblocked by https://github.com/ONSdigital/eq-survey-runner/pull/1998

File: app/subscriber.py [hunk 1]
```diff
@@ -34,16 +34,23 @@ def receipt_to_case(message: Message):
...
-                    object_name=object_name)
-        payload = json.loads(message.data)
-        time_obj_created = parse_datetime(payload['timeCreated']).isoformat()
-    except KeyError:
-        logger.exception('Pub/Sub Message missing attribute')
+        payload = json.loads(message.data)  # parse metadata as JSON payload
+        metadata = payload['metadata']
+        case_id, tx_id, created = metadata['case_id'], metadata['tx_id'], payload['timeCreated']
+    except KeyError as e:
+        logger.error('Pub/Sub Message missing required data', missing=e.args[0])
```
Comment (davidmort) [inline app/subscriber.py]: we prob should be logging something out here to enable us to link back to eq somehow

**Review by davidmort — DISMISSED** [review]

**Comment by davidmort** [comment]
just the one comment, otherwise all looks good :)

**Review by liamtoozer — APPROVED** [review]
Tested locally and looks good to me 👍

**Review by davidmort — APPROVED** [review]

**Pull request merged**